lib/smack/SmackRep.cpp

```diff
@@ -154,8 +191,8 @@ void SmackRep::collectRegions(llvm::Module &M) {
 
 const Stmt* SmackRep::alloca(llvm::AllocaInst& i) {  
   const Expr* size = 
-    Expr::fn("$mul",lit(storageSize(i.getAllocatedType())),lit(i.getArraySize()));
-                       
+    Expr::fn("$mul.ref", lit(storageSize(i.getAllocatedType()), width), SmackOptions::BitVectors? Expr::fn("$zext.i32.ref", lit(i.getArraySize())) : lit(i.getArraySize()));
```

Comment:
The argument to `alloca` has type `size`, not `ref`. Also, the width of pointers is irrelevant here. The width of `size` is what's important.